SSA/Projects/InstCombine/LLVM/Semantics.lean

```diff
@@ -4,23 +4,37 @@ import Mathlib.Tactic.Cases
 import Mathlib.Tactic.SplitIfs
 import Mathlib.Tactic.Tauto
 import SSA.Projects.InstCombine.LLVM.SimpSet
+import SSA.Core.Util
 
 
 namespace LLVM
 
+
+def IntN w := Option (BitVec w)
+
+def and??? {w: Nat} (x y : IntN w) : IntN w :=
+match x, y with
+| some xx, some yy => some (xx &&& yy)
+| _, _ => none
+
 /--
 The ‘and’ instruction returns the bitwise logical and of its two operands.
 -/
 @[simp_llvm]
-def and? {w : Nat} (x y : BitVec w) : Option <| BitVec w :=
+def and? {w : Nat} (x y : BitVec w) : IntN w :=
   some <| x &&& y
 
+def and?? {w : Nat} (x y : IntN w) : IntN w := do
```

Comment:
isn't it better to use `pairBind`? this definition style will end up with a lot of verbose repeated code

Reply:
I updated the tests to use a uniform style. Let me know what you think.